### ISSUE TITLE
0xR360 - DOS can be performed on EMPAM module settle function when the auction fills certain conditions

### ISSUE THREAD

**Issue by sherlock-admin4** [sherlock-audit/2024-03-axis-finance-judging#40]
0xR360

medium

# DOS can be performed on EMPAM module settle function when the auction fills certain conditions

## Summary
DOS can be performed on EMPAM module settle function by submitting many low value bids with minimum price to the Auction. 

## Vulnerability Detail
When an auction with a low enough minimum price is created, the settling can be blocked by a big amount of bids. The attacker would need to send enough bids with a bid that fills these conditions: 

- amount > auction.minBidSize * auction.minPrice / auction.baseTokenDecimals
- price > auction.minPrice
- amountOut > auction.minBidSize
- the auction must not be filled by other users. This is expected by the protocol since is an order book (moved by supply/demand).

The only investment for the attacker is "amount" since he needs to transfer on bid creation. If a user sets an auction.minPrice low enough to allow many small bids without filling the auction, an attacker can deny the call to "settle". The protocol only enforces a min price above 0.

When a bid is created and decrypted it fills an [array](https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L506). This array is [looped through](https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L610-L611) on [_getLotMarginalPrice](https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L595)  called on [settling](https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L758). With enough bids this operation can run out of gas and never succeed.

## Impact
Loss of user funds deposited on the auction. Loss of other users who might also try to fill the auction. Only recoverable if filled completely by another user, if the price of the auction is too far from current market price the loss will incur in the user who filled it and the attacker will recover his funds.

## Code Snippet

https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L610-L611
https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/src/modules/auctions/EMPAM.sol#L758

This can be tested by putting the following code on the [settle tests for EMPA](https://github.com/sherlock-audit/2024-03-axis-finance/blob/main/moonraker/test/modules/auctions/EMPA/settle.t.sol).

```solidity
     modifier audit_givenBidsDOS() {
        uint bidsNum = 14000;

        for (uint i=0;i < bidsNum; i++){
            _createBid(
                _scaleQuoteTokenAmount(_BID_PRICE_ONE_AMOUNT),
                _scaleBaseTokenAmount(_BID_PRICE_ONE_AMOUNT_OUT)
            );
        }

        _;
    }

    function test_audit_dos_on_failed_fill()
        external
        givenLotIsCreated
        givenLotHasStarted
        audit_givenBidsDOS
        givenLotHasConcluded
        givenPrivateKeyIsSubmitted
        givenLotIsDecrypted
    {
        uint256 gasBefore = gasleft();
        // best case against the attacker is max gas limit on block and that its the only tx on it.
        uint ETH_BLOCK_GAS_LIMIT = 30_000_000; 
        (Auction.Settlement memory settlement, bytes memory auctionOutput) = _settle();
         uint256 gasUsed = gasBefore -  gasleft();
        //gas used without taking in account the start of the tx and the entripoint on auction house
        assertEq(gasUsed > ETH_BLOCK_GAS_LIMIT,true, "out of gas");
    }
```

## Tool used

Manual Review

## Recommendation

- Make the settle function on this module called in batches or
- Enforce a relation between the minPrice and minBidSize such that the condition
amount > auction.minBidSize * auction.minPrice / auction.baseTokenDecimals
holds true for high enough "amounts" that will fill the auction in few bids

Duplicate of #237
